### PR TITLE
enh(UI): Add tooltip for errors messages during ba-bv's configuration

### PR DESCRIPTION
**Pull request by Jordan91**
How to reproduce:

Open the Business Activity Add panel
Go on the “name” input, add a character, remove it, so the “required” label displayed, but shifts elements on the line.

Expected result : Align the elements or add tooltip to display the error message

### CODE REVIEW

File: packages/centreon-ui/src/InputField/Text/index.tsx [hunk 1]
```diff
@@ -81,6 +84,16 @@ const TextField = React.forwardRef(
   ): JSX.Element => {
     const classes = useStyles();
 
+    const ErrorLabel = error;
+    const getTooltipErrorLabel = always(error);
+
+    const headerContent = (
+      <Tooltip placement="top" title={getTooltipErrorLabel(ErrorLabel)}>
```
Comment (Thebarda) [inline packages/centreon-ui/src/InputField/Text/index.tsx:91]:
```suggestion
      <Tooltip placement="top" title={error || ''}>
```

File: packages/centreon-ui/src/InputField/Text/index.tsx [hunk 2]
```diff
@@ -81,6 +84,16 @@ const TextField = React.forwardRef(
   ): JSX.Element => {
     const classes = useStyles();
 
+    const ErrorLabel = error;
+    const getTooltipErrorLabel = always(error);
```
Comment (Thebarda) [inline packages/centreon-ui/src/InputField/Text/index.tsx:88]:
```suggestion
```

File: packages/centreon-ui/src/InputField/Text/index.tsx [hunk 3]
```diff
@@ -1,16 +1,19 @@
 import React from 'react';
 
 import clsx from 'clsx';
-import { equals, isNil, not } from 'ramda';
+import { always, equals, isNil, not } from 'ramda';
```
Comment (Thebarda) [inline packages/centreon-ui/src/InputField/Text/index.tsx:4]:
```suggestion
import { equals, isNil, not } from 'ramda';
```

File: packages/centreon-ui/src/InputField/Text/index.tsx [hunk 4]
```diff
@@ -81,43 +84,53 @@ const TextField = React.forwardRef(
   ): JSX.Element => {
     const classes = useStyles();
 
+    const headerContent = (
+      <Tooltip placement="top" title={error || ''}>
+        <div>
+          <HeaderLabel>{error}</HeaderLabel>
+        </div>
+      </Tooltip>
+    );
```
Comment (Thebarda) [inline packages/centreon-ui/src/InputField/Text/index.tsx:93]: You don't need this anymore
```suggestion
```

File: packages/centreon-ui/src/InputField/Text/index.tsx [hunk 5]
```diff
@@ -81,43 +84,53 @@ const TextField = React.forwardRef(
...
-            [classes.compact]: isSizeEqualTo(Size.compact),
-          }),
-        }}
-        label={label}
-        ref={ref}
-        size="small"
-        variant="filled"
-        {...rest}
-      />
+      <Tooltip title={error as string}>
```
Comment (Thebarda) [inline packages/centreon-ui/src/InputField/Text/index.tsx:97]:
```suggestion
      <Tooltip title={error || ''}>
```

File: packages/centreon-ui/src/InputField/Text/index.tsx [hunk 6]
```diff
@@ -81,43 +84,53 @@ const TextField = React.forwardRef(
...
+                <EndAdornment />
+              </OptionalLabelInputAdornment>
+            ),
+            startAdornment: StartAdornment && (
+              <OptionalLabelInputAdornment label={label} position="start">
+                <StartAdornment />
+              </OptionalLabelInputAdornment>
+            ),
+          }}
+          {...headerContent}
```
Comment (Thebarda) [inline packages/centreon-ui/src/InputField/Text/index.tsx:115]:
```suggestion
```

File: packages/centreon-ui/src/InputField/Text/index.tsx [hunk 7]
```diff
@@ -9,8 +9,11 @@ import {
   TextFieldProps,
   Theme,
   makeStyles,
+  Tooltip,
 } from '@material-ui/core';
 
+import HeaderLabel from '../../Listing/Header/Label';
+
```
Comment (Thebarda) [inline packages/centreon-ui/src/InputField/Text/index.tsx:16]: aaaand you don't need this anymore :)
```suggestion
```

**Comment by bdauria** [comment]
@Jordan91 I don't think we should change the default behaviour of displaying the error message in the helper text. We should add the option of displaying it as a tooltip instead.

**Review by jdelpierre — APPROVED** [review]

**Review by Thebarda — APPROVED** [review]

**Pull request merged**